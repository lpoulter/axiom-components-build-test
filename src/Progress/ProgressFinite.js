import PropTypes from 'prop-types';
import React, { Component } from 'react';
import renderFilter from './_filter';
import { getArcProperties } from './utils';
import RadialProgress from './RadialProgress';
import './ProgressFinite.css';

export default class ProgressFinite extends Component {
  static propTypes = {
    /** Percentage of progress complete */
    percent: PropTypes.number.isRequired,
    /** Size of the indicator */
    size: PropTypes.oneOf(['small', 'medium', 'large']),
    /** Size in REM units */
    sizeRem: PropTypes.string,
  };

  static defaultProps = {
    percent: 0,
    size: 'small',
  }

  UNSAFE_componentWillMount = renderFilter

  render() {
    const {
      percent,
      ...rest
    } = this.props;

    const {
      diameter,
      innerRadius,
      strokeWidth,
      strokeDasharray,
      strokeDashoffset,
    } = getArcProperties({ percent });

    return (
      <RadialProgress { ...rest } className="ax-progress-finite" diameter={ diameter }>
        <circle
            className="ax-progress-finite__background"
            r={ innerRadius }
            style={ { strokeWidth } } />
        <circle
            className="ax-progress-finite__arc"
            r={ innerRadius }
            style={ { strokeDasharray, strokeDashoffset, strokeWidth } } />
      </RadialProgress>
    );
  }
}
