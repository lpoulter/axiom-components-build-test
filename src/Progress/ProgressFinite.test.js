import React from 'react';
import renderer from 'react-test-renderer';
import ProgressFinite from './ProgressFinite';

const getComponent = (props = {}) =>
  renderer.create(
    <ProgressFinite { ...props } />
  );

describe('ProgressFinite', () => {
  it('renders with defaultProps', () => {
    const component = getComponent();
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with percent props', () => {
    const component = getComponent({ percent: 50 });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
