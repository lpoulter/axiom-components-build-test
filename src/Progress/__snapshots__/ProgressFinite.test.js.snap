// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressFinite renders with defaultProps 1`] = `
<svg
  className="ax-progress-finite ax-radial-progress ax-radial-progress--small"
  style={Object {}}
  viewBox="0 0 24 24"
>
  <g
    transform="translate(12, 12) rotate(-90)"
  >
    <circle
      className="ax-progress-finite__background"
      r={10}
      style={
        Object {
          "strokeWidth": 4,
        }
      }
    />
    <circle
      className="ax-progress-finite__arc"
      r={10}
      style={
        Object {
          "strokeDasharray": 62.83185307179586,
          "strokeDashoffset": 62.83185307179586,
          "strokeWidth": 4,
        }
      }
    />
  </g>
</svg>
`;

exports[`ProgressFinite renders with percent props 1`] = `
<svg
  className="ax-progress-finite ax-radial-progress ax-radial-progress--small"
  style={Object {}}
  viewBox="0 0 24 24"
>
  <g
    transform="translate(12, 12) rotate(-90)"
  >
    <circle
      className="ax-progress-finite__background"
      r={10}
      style={
        Object {
          "strokeWidth": 4,
        }
      }
    />
    <circle
      className="ax-progress-finite__arc"
      r={10}
      style={
        Object {
          "strokeDasharray": 62.83185307179586,
          "strokeDashoffset": 31.41592653589793,
          "strokeWidth": 4,
        }
      }
    />
  </g>
</svg>
`;
